### ISSUE TITLE
Using the EsSpark.esRDD method to read from Elasticsearch does not honor size parameters in either URI or DSL

### ISSUE THREAD

**Issue by sherry-ger** [elastic/elasticsearch-hadoop#469]
The size option is ignored.  Both methods returned all results.

```
//conf.set("es.query", "?q=text_entry:rebel&size=1") or

val q1 = "{\"query\": {\"filtered\" : {\"query\" : {\"term\" : { \"text_entry\": \"rebel\" }}}},\"size\" : 1}"
conf.set("es.query", q1)
val esRDD = sc.newAPIHadoopRDD(conf, classOf[EsInputFormat[Text, MapWritable]], classOf[Text], classOf[MapWritable])
```

May be this is similar to #444

**Comment by costin** [comment]
This is actually on purpose and needs to be documented. Since the connector does a parallel query, it also looks at the number of documents being returned so if the user specifies a parameter, it will overwrite it according to the `batch.size` setting (see the configuration option).
In other words, if you want to control the size, do so through that setting as it will always take precedence.

**Comment by jaysonminard** [comment]
its nice to have docs...

**Comment by costin** [comment]
relates #546

**Comment by costin** [comment]
Fixed through #546
